lib/auth/trustedcluster.go

```diff
@@ -35,41 +35,49 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// UpsertTrustedCluster will either create a new services.TrustedCluster or
-// can be used to toggle a services.TrustedCluster on and off.
+// UpsertTrustedCluster changes the state of trust relationship. If we have do
+// not have an exisiting trusted cluster resource in the backend a new one will
+// be created (either enabled or disabled). If we do, we will enable to disable
+// the existing resource.
 func (a *AuthServer) UpsertTrustedCluster(t services.TrustedCluster) error {
-	// check if the trusted cluster exists already
 	var exists bool
-	_, err := a.Presence.GetTrustedCluster(t.GetName())
+	existingCluster, err := a.Presence.GetTrustedCluster(t.GetName())
 	if err == nil {
 		exists = true
 	}
+	enable := t.GetEnabled()
 
-	// if the trusted cluster does not exist, do the initial token exchange
-	// to establish trust relationship and return.
-	if !exists {
-		err = a.establishTrust(t)
+	// if we are not making any changes, return nil right away
+	if exists == true && existingCluster.GetEnabled() == enable {
```

Comment:
`exists` seems to be redundant, could be `if existingCluster !=nil && existingCluster.GetEnabled()`